misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -239,46 +212,178 @@ private static void generateRecords(Element xmlElement, boolean isClosed,
+        org.w3c.dom.NodeList xmlNodeList = xmlElement.getChildNodes();
+        for (int i = 0; i < xmlNodeList.getLength(); i++) {
+            org.w3c.dom.Node xmlNode = xmlNodeList.item(i);
+
+            if (xmlNode.getNodeType() == org.w3c.dom.Node.ELEMENT_NODE) {
+                Element xmlElementNode = (Element) xmlNode;
+                boolean isLeafXMLElementNode = isLeafXMLElementNode(xmlElementNode);
+                if (!isLeafXMLElementNode) {
+                    generateRecords(xmlElementNode, isClosed, recordToTypeDescNodes, recordToAnnotationNodes,
+                            recordToElementNodes, diagnosticMessages);
+                }
+                RecordFieldNode recordField = getRecordField(xmlElementNode, false);
+                if (recordFields.stream().anyMatch(recField -> ((RecordFieldNode) recField).fieldName().text()
+                        .equals(recordField.fieldName().text()))) {
+                    int indexOfRecordFieldNode = IntStream.range(0, recordFields.size())
+                            .filter(j -> ((RecordFieldNode) recordFields.get(j)).fieldName().text()
+                                    .equals(recordField.fieldName().text())).findFirst().orElse(-1);
+                    if (indexOfRecordFieldNode == -1) {
+                        recordFields.add(recordField);
+                    } else {
+                        RecordFieldNode existingRecordField =
+                                (RecordFieldNode) recordFields.remove(indexOfRecordFieldNode);
+                        RecordFieldNode updatedRecordField = mergeRecordFields(existingRecordField, recordField);
+                        recordFields.add(indexOfRecordFieldNode, updatedRecordField);
+                    }
+                } else {
+                    recordFields.add(recordField);
+                }
+            }
+        }
+        org.w3c.dom.NamedNodeMap xmlAttributesMap = xmlElement.getAttributes();
+        for (int i = 0; i < xmlAttributesMap.getLength(); i++) {
+            org.w3c.dom.Node xmlNode = xmlAttributesMap.item(i);
+            if (xmlNode.getNodeType() == org.w3c.dom.Node.ATTRIBUTE_NODE) {
+                if (xmlElement.getPrefix() != null &&
+                        xmlNode.getPrefix() != null &&
+                        xmlNode.getPrefix().equals(XMLNS_PREFIX) &&
+                        xmlElement.getPrefix().equals(xmlNode.getLocalName())) {
+                    AnnotationNode xmlNSNode = getXMLNamespaceNode(xmlNode.getLocalName(), xmlNode.getNodeValue());
+                    recordToAnnotationNodes.put(xmlNodeName, xmlNSNode);
+                } else {
+                    Node recordField = getRecordField(xmlNode);
+                    recordFields.add(recordField);
+                }
+            }
+        }
+
+        return recordFields;
+    }
+
+    private static List<Node> updateRecordFields(Map<String, RecordFieldNode> previousRecordFieldToNodes,
```

Comment:
can you add documentation/comments explaining the logic here, as the name `updateRecordFields` seem too generic?